index.js

```diff
@@ -61,50 +62,58 @@ export class SparqlHandler {
 
   async exists (iri, query) {
     debug('SPARQL exists query for IRI <' + iri + '> : ' + query)
-
-    const res = await this.client.selectQuery(query, this.buildQueryOptions())
-    const status = res.status
-    if (status !== 200) {
-      return { status, undefined }
+    try {
+      const exists = await this.parsingClient.query.ask(query,
+        this.buildQueryOptions())
+      return { exists, status: 200 }
+    } catch (error) {
+      return { status: error.status }
     }
-    const json = await res.json()
-    const exists = json.boolean
-    return { status, exists }
   }
 
   async graphStream (iri, query, accept) {
     debug('SPARQL query for IRI <' + iri + '> : ' + query)
 
-    const queryOptions = this.buildQueryOptions()
+    const headers = this.buildQueryOptions()
+    headers.accept = accept
 
-    queryOptions.accept = accept
-
-    const res = await this.client.constructQuery(query, queryOptions)
-    if (res.status !== 200) {
-      return { status: res.status }
-    }
-    const headers = {}
-
-    res.headers.forEach((value, name) => {
-      // stream will be decoded by the client -> remove content-encoding header
-      if (name === 'content-encoding') {
-        return
-      }
-      headers[name] = value
-    })
+    const response = await this.simpleClient.query.construct(query, { headers })
     return {
-      status: res.status,
-      headers: headers,
-      stream: res.body
+      status: response.status, stream: response.body
     }
   }
 
   handle (req, res, next) {
-    if (req.method === 'GET') {
-      this.get(req, res, next, req.iri)
-    } else {
-      next()
+    switch (req.method) {
+      case 'HEAD':
+        return this.head(req, res, next, req.iri)
+      case 'GET':
+        return this.get(req, res, next, req.iri)
     }
+
+    return next()
+  }
+
+  async head (_req, res, next, iri) {
+    iri = encodeURI(iri)
+
+    debug('handle OPTIONS request for IRI <' + iri + '>')
```

Comment:
I noticed that I forgot to replace this `OPTIONS` by `HEAD` in this log line.
Can you take care of it? :)

```suggestion
    debug('handle HEAD request for IRI <' + iri + '>')
```

Reply:
Sure!, I've updated it now.

Reply:
Updated!